### PR TITLE
Add main, config, provider

### PR DESCRIPTION
**Pull request by jtsaito**
I re-wrote the PR on request by @jansiwy to make reviewing easier. Each previous commit is now a separate PR. The PRs are building up on top of each other.

This PR is for reviewing only.

- https://github.com/babbel/terraform-provider-travisci/pull/15 <- you are here
- https://github.com/babbel/terraform-provider-travisci/pull/16
- https://github.com/babbel/terraform-provider-travisci/pull/18
- https://github.com/babbel/terraform-provider-travisci/pull/19
- https://github.com/babbel/terraform-provider-travisci/pull/20

### CODE REVIEW

File: travisci/provider.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+				DefaultFunc: schema.EnvDefaultFunc("GITHUB_OWNER", nil),
+				Description: descriptions["github_owner"],
+			},
+			"travis_api_url": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("TRAVIS_API_URL", travis.TRAVIS_API_DEFAULT_URL),
+				Description: descriptions["travis_api_url"],
+			},
+			"travisci_token": &schema.Schema{
```
Comment (fmasuhr) [inline travisci/provider.go]: I suggest to unify the naming `travis` vs `travis-ci`

File: travisci/config.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package travisci
+
+import travis "github.com/jtsaito/go-travis"
+
+// Config contains information necessary for creating the Travis-CI client
+type Config struct {
+	GithubOwner   string // github organisation or user
+	TravisAPIURL  string // travis api (pro or default)
```
Comment (jansiwy) [inline travisci/config.go]: please unify travis vs. travis ci

File: travisci/config.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	TravisciToken string // private travis token
+}
+
+// Client configures and returns a fully initialized Travis-CI client
+func (c *Config) Client() (*travis.Client, error) {
+	tc := travis.NewClient(c.TravisAPIURL, "")
```
Comment (jansiwy) [inline travisci/config.go]: why is there an empty string?

File: travisci/provider.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+			},
+		},
+		ConfigureFunc: providerConfigure,
+	}
+}
+
+var descriptions map[string]string
+
+func init() {
+	descriptions = map[string]string{
```
Comment (jansiwy) [inline travisci/provider.go]: please inline until you need to reuse those descriptions

**Comment by jtsaito** [comment]
closing review branch: outdated

**Pull request closed without merging**